File: javascript/pe025.js
"use strict"

const limit = 1000;

function fib(n, vector)
{
  // this computes the fibonacci number of the passed number
  if (n < 2)
    vector[vector.length] = n;
  else
    vector[vector.length] = BigInt(vector[n - 2]) + BigInt(vector[n - 1]);
}

let i = 0;
let vector = [];
fib(i, vector);
while (String(vector[vector.length - 1]).length < limit)
{
  i += 1;
  fib(i, vector);
}

console.log(i)

File: python/pe025.py
#!/usr/bin/env python3

""" this computes projecteuler.net problem 025 """

import sys


def fib(n, vector):
    """ this computes the fibonacci number of the passed number """
    if n < 2:
        vector.append(n)
    else:
        vector.append(vector[n - 2] + vector[n - 1])


LIMIT = 1000
i = 0
vector = []
fib(i, vector)
while len(str(vector[i])) < LIMIT:
    i += 1
    fib(i, vector)

print(i)
sys.exit(0)
